mymodule/sky_sim.py

```diff
@@ -23,20 +23,48 @@ def get_radec():
     return ra, dec
 
 
+def crop_to_circle(ras,decs, ref_ra, ref_dec, radius):
+    """
+    Crop an input list of positions so that they lie within radius of
+    a reference position
+
+    Parameters
+    ----------
+    ras,decs : list(float)
+        The ra and dec in degrees of the data points
+    ref_ra, ref_dec: float
+        The reference location
+    radius: float
+        The radius in degrees
+    Returns
+    -------
+    ras, decs : list
+        A list of ra and dec coordinates that pass our filter.
+    """
+    ra_out = []
+    dec_out = []
+    for i in range(len(ras)):
+        dist_to_ref = sqrt((ra - ras[i])**2 + (dec - decs[i])**2)
+        if dist_to_ref < radius
+        ra_out.append(ras[i])
+        dec_out.append(decs[i])
+    return ra_out, dec_out
+
+
 def make_stars(ra, dec, nsrc=NSRC):
     ras = []
     decs = []
     for _ in range(nsrc):
         ras.append(ra + random.uniform(-1, 1))
-        decs.append(dec + random.uniform(-1, 1))
+        decs.append(dec + random.uniform(-1, 1))   
+    # applying the filter
+    ras, decs = crop_to_circle(ras,decs)    
```

Comment:
This cropping will result in fewer sources, so that less than nsrc sources are returned